plugins/cxxrandom.cpp

```diff
@@ -137,118 +123,100 @@ class NumberSequence
     std::vector<int64_t> m_numbers;
 public:
     NumberSequence(){}
-    NumberSequence( int64_t start, int64_t end )
-    {
-        for( int64_t i = start; i <= end; ++i )
-        {
+    NumberSequence( int64_t start, int64_t end ) {
+        for( int64_t i = start; i <= end; ++i ) {
             m_numbers.push_back( i );
         }
     }
     void Add( int64_t num ) { m_numbers.push_back( num ); }
-    void Reset()            { m_numbers.clear(); }
-    int64_t Next()
-    {
-        if(m_position >= m_numbers.size())
-        {
+    void Reset() { m_numbers.clear(); }
+    int64_t Next() {
+        if(m_position >= m_numbers.size()) {
             m_position = 0;
         }
         return m_numbers[m_position++];
     }
-    void Shuffle( uint16_t id )
-    {
-        std::shuffle( std::begin( m_numbers ), std::end( m_numbers ), EnginesKeeper::Instance().RNG( id ) );
+    void Shuffle( uint16_t engID ) {
+        std::shuffle( std::begin( m_numbers ), std::end( m_numbers ), EnginesKeeper::Instance().RNG(engID));
     }
-    void Print()
-    {
-        for( auto v : m_numbers )
-        {
+    void Print() {
+        for( auto v : m_numbers ) {
             cout->print( "%" PRId64 " ", v );
         }
     }
 };
 
+#define SK_ID_BASE 0
+
 class SequenceKeeper
 {
 private:
     SequenceKeeper() {}
     std::unordered_map<uint16_t, NumberSequence> m_sequences;
-    uint16_t counter = 0;
+    uint16_t id_counter = SK_ID_BASE;
 public:
-    static SequenceKeeper& Instance()
-    {
+    static SequenceKeeper& Instance() {
         static SequenceKeeper instance;
         return instance;
     }
-    uint16_t MakeNumSequence( int64_t start, int64_t end )
-    {
-        m_sequences[++counter] = NumberSequence( start, end );
-        return counter;
+    uint16_t MakeNumSequence( int64_t start, int64_t end ) {
+        m_sequences[++id_counter] = NumberSequence(start, end);
```

Comment:
Shouldn't there be some collision detection here? Ids are only 16 bits after all.  Also for NewEngine above

Reply:
I don't see it as a big concern. The ids would overlap the other's range before they'd collide with themselves. In either case though, I don't see much that can be done without some overboard complexity for the problem.

Reply:
How about just testing whether the id already exists in the map and erroring out if it's there?

Reply:
Sure, doable. Simple safety check. It'll make more sense if I also increase the int sizes and or do the identity thingy